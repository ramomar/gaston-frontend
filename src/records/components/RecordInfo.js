import React from 'react';
import Shapes from '../shapes';
import { Text, Box } from 'grommet';
import * as R from 'ramda';

function RecordInfoItem(props) {
  const { fieldName, fieldValue } = props;

  return (
    <Box direction='column' margin={{ bottom: 'large'}} style={{ fontFamily: 'Courier New, monospace' }}>
      <Box margin={{ bottom: 'small' }}>
          <Text weight='bold'>{fieldName}</Text>
      </Box>
      <Box>
          <Text>{fieldValue}</Text>
      </Box>
    </Box>
  );
}

function RecordInfo(props) {
  const { record } = props;
  const items = R.toPairs(JSON.parse(record.raw));

  return (
    <Box direction='column' fill='horizontal' flex='grow' pad='small'>
      <Box margin={{ vertical: 'xlarge' }}>
        <Text weight='bold' size='large' alignSelf='center'>
          Detalles
        </Text>
      </Box>
      {items.map(([key, value]) => <RecordInfoItem key={key} fieldName={key} fieldValue={JSON.stringify(value)} />)}
    </Box>
  );
}

RecordInfo.propTypes = {
  record: Shapes.record.isRequired
};

export default RecordInfo;
