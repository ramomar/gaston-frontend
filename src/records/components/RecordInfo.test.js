import React from 'react';
import renderer from 'react-test-renderer';
import RecordInfo from './RecordInfo';
import { DateTime } from 'luxon';

it('renders correctly', () => {
    const record = {
        id: '0007182d-54cb-42b7-88fc-bbaba51db198',
        amount: 2000.00,
        date: DateTime.fromISO('2021-01-17T12:38:51.700Z'),
        note: 'RETIRO DE EFECTIVO',
        raw: '{"source": "CASH_WITHDRAWAL_EMAIL", "type": "EXPENSE", "note": "RETIRO DE EFECTIVO", "amount": "2000.00", "operation_date": "17/Feb/2021 12:38:51 HORAS", "application_date": "17/Feb/2021", "receiver": null, "channel": {"type": "Cajero Banorte", "details": {"name": "ROMA SUR 4", "location": "CDMX"}}, "extra_amounts": []}'
    };

    const component = <RecordInfo record={record} />;

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
});
