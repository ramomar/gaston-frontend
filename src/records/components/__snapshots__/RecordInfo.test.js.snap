// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 bmEWzQ"
>
  <div
    className="StyledBox-sc-13pk1d4-0 diNbmr"
  >
    <span
      className="StyledText-sc-1sadyjn-0 dvuZpX"
      size="large"
    >
      Detalles
    </span>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        source
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        "CASH_WITHDRAWAL_EMAIL"
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        type
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        "EXPENSE"
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        note
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        "RETIRO DE EFECTIVO"
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        amount
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        "2000.00"
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        operation_date
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        "17/Feb/2021 12:38:51 HORAS"
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        application_date
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        "17/Feb/2021"
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        receiver
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        null
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        channel
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        {"type":"Cajero Banorte","details":{"name":"ROMA SUR 4","location":"CDMX"}}
      </span>
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 hMaZmZ"
    style={
      Object {
        "fontFamily": "Courier New, monospace",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 lfluzk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 bnqisW"
      >
        extra_amounts
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fWrrBh"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        []
      </span>
    </div>
  </div>
</div>
`;
